### PR TITLE
CAMEL-14626 : api maven plugin - Cannot load <T> class

### PR DESCRIPTION
**Pull request by manonthegithub**
refs #CAMEL-14626

### CODE REVIEW

File: tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java [hunk 1]
```diff
@@ -344,9 +344,15 @@ public String getCanonicalName(ApiMethodArg argument) throws MojoExecutionExcept
                         parameterizedType.append(
                             getCanonicalName(getProjectClassLoader().loadClass("java.lang." + argType)));
                     } catch (ClassNotFoundException e1) {
-                        log.warn("Ignoring type parameters <" + typeArgs + "> for argument " + argument.getName()
-                                 + ", unable to load parametric type argument " + argType, e1);
-                        ignore = true;
+                        // if the length of the artType is 1, we think that it's variable type parameter (like T in List<T>)
+                        // not perfect solution, but should work in most of the cases
+                        if (argType.trim().length() > 1) {
```
Comment (manonthegithub) [inline tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: Could also do it at line 335, but I think this approach is more universal.

**Comment by manonthegithub** [comment]
Currently, if there is a class in type args which is not found, parameters are ignored and generic class is substitute with a raw class, i.e. `List<T> -> List` or `Map<String, T> -> Map`

**Comment by manonthegithub** [comment]
Build failure is due: Maven repo download failed of one of the deps
shouldnt be a problem

File: tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java (hunk 1 above)
Comment (dmvolod) [inline tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: @manonthegithub I'm not sure, but we could try to implement more elegant solution, like cast to (T) and catch ClassCastException?

Comment (manonthegithub) [reply tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: @dmvolod it is already checked that the clas does not exits on the line 335 and then 344, how would additional check help?

Comment (manonthegithub) [reply tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: Didn't do much of class casting, but I think it's impossible to cast having only string value of a class name.

Comment (manonthegithub) [reply tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: * I mean a String in runtime with the name of type arg

Comment (dmvolod) [reply tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: Yes, you are right, but looks like not handles all generics variants. But may be it's enough for current cases.

Comment (manonthegithub) [reply tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java:349]: That is true, but it's not really possible to distinguish here between Generic parameter and nonexistent class.

**Comment by manonthegithub** [comment]
Please do not merge, I will do some changes later.

**Comment by davsclaus** [comment]
Okay thanks, let us know when its ready.

**Comment by manonthegithub** [comment]
@davsclaus Now should be fine.

**Pull request merged**